test/test_helper.rb

```diff
@@ -28,6 +28,12 @@ def assert_logs_match(regexp)
       @logger_stream.rewind
       assert_match regexp, @logger_stream.read
     end
+
+    def assert_logs_match_once(regexp)
+      @logger_stream.rewind
+      count = @logger_stream.read.scan(regexp).count
+      assert_equal 1, count, "Expected #{regexp} to appear only once in the log, appear #{count} times"
+    end
```

Comment:
Instead of introducing a new helper, I think you should consider adding a `times` argument to `assert_logs_match`. That's more similar to helpers we have elsewhere internally :)